Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -483,7 +484,27 @@ namespace AZ
         void SwapChain::InvalidateNativeSwapChain()
         {
             auto& device = static_cast<Device&>(GetDevice());
-            auto presentCommand = [this,  &device]([[maybe_unused]] void* queue)
+            auto presentCommand = [this, &device]([[maybe_unused]] void* queue)
+            {
+                vkDeviceWaitIdle(device.GetNativeDevice());
+                if (m_nativeSwapChain != VK_NULL_HANDLE)
```

Comment:
We cant as we need to pass the old swapchain to the drivers when creating the new swapchain (which happens right after this function). This ensures that the present call for the current frame does not yield VK_ERROR_OUT_OF_DATE_KHR and force us to then recreate again. Basically we would get a resize event and we would recreate the swapchain and then recreate again as the swapchain present would fail for the current frame.

Reply:
FlushCommands is here to ensure that the Vk calls above are executed on the presentation thread before this function exits. Basically any Vk commands related to Presentation we have been pushing on the presentation queue as the drivers do not like it when two different threads (main thread and presentation thread) are making presentation specific vk commands.